nens_auth_client/backends.py

```diff
@@ -111,4 +111,49 @@ def authenticate(self, request, claims):
         # Create a permanent association
         create_remote_user(user, claims)
 
-        return user if self.user_can_authenticate(user) else None
+        return user
+
+
+class AcceptNensBackend(ModelBackend):
+    def authenticate(self, request, claims):
+        """Backend for auto-accepting users that have a N&S azure AD account.
+
+        The behaviour looks a bit like the SSOMigrationBackend above, but with
+        two key differences:
+
+        - N&S users don't need an existing account, they're accepted right
+          away and a User is created if missing.
+
+        - We only deal with users recognized as being really from N&S.
+
+        Args:
+          request: the current request
+          claims (dict): the verified payload of the ID or Access token
+
+        Returns:
+          user or None
+
+        """
+        username = _nens_user_extract_username(claims)
+        if username is None:
+            return
+
+        email = claims.get("email")
+
+        try:
+            user = UserModel.objects.get(username__iexact=username, email__iexact=email)
```

Comment:
Mooi! Ik zat hier nog even over na te denken. Er zijn een paar users (zoals `jonas.vanschrojenste` van wie de username op 20 characters worden geclipt. Dan werkt dit hier niet.

Misschien is  voor applicaties als TRS alleen een match op email voldoende veilig? Als emails in de `User` database staan, zijn ze hoe dan ook geverifieerd namelijk (zowel via SSO als via Cognito)


```suggestion
            user = UserModel.objects.get(email__iexact=email)
```

Reply:
Daar heb je wel gelijk in. 0.01% minder veilig en 20% minder kans op rare Jonas-vormige corner cases => 19.99% verbetering!

Reply:
Gedaan.